extras/deepstream/deepstream-sample/nvdsparsebbox_retinanet.cpp

```diff
@@ -0,0 +1,118 @@
+  static int bboxLayerIndex = -1;
+  static int classesLayerIndex = -1;
+  static int scoresLayerIndex = -1;
+  static NvDsInferDimsCHW scoresLayerDims;
+  int numDetsToParse;
+
+  /* Find the bbox layer */
+  if (bboxLayerIndex == -1) {
+    for (unsigned int i = 0; i < outputLayersInfo.size(); i++) {
+      if (strcmp(outputLayersInfo[i].layerName, "boxes") == 0) {
+        bboxLayerIndex = i;
+        break;
+      }
+    }
+    if (bboxLayerIndex == -1) {
+    std::cerr << "Could not find bbox layer buffer while parsing" << std::endl;
+    return false;
+    }
+  }
+
+  /* Find the scores layer */
+  if (scoresLayerIndex == -1) {
+    for (unsigned int i = 0; i < outputLayersInfo.size(); i++) {
+      if (strcmp(outputLayersInfo[i].layerName, "scores") == 0) {
+        scoresLayerIndex = i;
+        getDimsCHWFromDims(scoresLayerDims, outputLayersInfo[i].dims);
+        break;
+      }
+    }
+    if (scoresLayerIndex == -1) {
+    std::cerr << "Could not find scores layer buffer while parsing" << std::endl;
+    return false;
+    }
+  }
+
+  /* Find the classes layer */
+  if (classesLayerIndex == -1) {
+    for (unsigned int i = 0; i < outputLayersInfo.size(); i++) {
+      if (strcmp(outputLayersInfo[i].layerName, "classes") == 0) {
+        classesLayerIndex = i;
+        break;
+      }
+    }
+    if (classesLayerIndex == -1) {
+    std::cerr << "Could not find classes layer buffer while parsing" << std::endl;
+    return false;
+    }
+  }  
+
+  
+  /* Calculate the number of detections to parse */
+  numDetsToParse = scoresLayerDims.c;
+  //std::cout << "DEBUG: Number of detections: " << numDetsToParse << std::endl;
+
+  float *bboxes = (float *) outputLayersInfo[bboxLayerIndex].buffer;
+  float *classes = (float *) outputLayersInfo[classesLayerIndex].buffer;
+  float *scores = (float *) outputLayersInfo[scoresLayerIndex].buffer;
+  // std::cout << "DEBUG: outputLayersInfo[scoresLayerIndex].buffer: " << outputLayersInfo[scoresLayerIndex].buffer << std::endl;
+  
+  for (int indx = 0; indx < numDetsToParse; indx++)
+  {
+    float outputX1 = bboxes[indx * 4];
+    float outputY1 = bboxes[indx * 4 + 1];
+    float outputX2 = bboxes[indx * 4 + 2];
+    float outputY2 = bboxes[indx * 4 + 3];
+    float this_class = classes[indx];
+    float this_score = scores[indx];
+    float threshold = detectionParams.perClassThreshold[this_class];
+    
+    if (this_score >= threshold)
+    {
+      NvDsInferParseObjectInfo object;
+      
+      object.classId = this_class;
+      object.detectionConfidence = this_score;
+
+      object.left = outputX1;
+      object.top = outputY1;
+      object.width = outputX2 - outputX1;
+      object.height = outputY2 - outputY1;
+
+      objectList.push_back(object);
+    }
+  }
+  //std::cout << "Returning from function" <<std::endl;
+  return true;
+}
+
+/* Check that the custom function has been defined correctly */
+CHECK_CUSTOM_PARSE_FUNC_PROTOTYPE(NvDsInferParseCustomRetinaNet);
```

Comment:
A mouthful, maybe we can find a simpler name?

Reply:
Would `NvDsInferParseRetinanet` work for you?

Reply:
👍